bugsnag-android-core/src/main/java/com/bugsnag/android/DeviceData.java

```diff
@@ -164,6 +164,9 @@ private boolean isRooted() {
      */
     private boolean isEmulator() {
         String fingerprint = Build.FINGERPRINT;
+        if (fingerprint == null) {
```

Comment:
Is this a case that should be added here, or in a separate PR?

Reply:
`android.os.Build` is an Android framework class, which means the return value of `Build.FINGERPRINT` will be null when run in a JVM unit test. Checking for null passed the test without making larger changes to the existing code.

Another option would be to collect all the existing fields in this class such as `Build` and to pass them in within a wrapper class that can be populated in a JVM test. For example, something like:

```
class BuildInfo {
    val fingerprint: String
    val manufacturer: String
}
```

I can make this change if it'd be preferable

Reply:
I have removed the null check for `fingerprint` and have instead created a new `DeviceBuildInfo` class which encapsulates all the static fields accessed from `android.os.Build`. This class is injected as a constructor parameter into `DeviceData` which allows for it to be swapped out with suitable values in unit tests.

The serialization tests have been updated as more fields are now serialized, now that the `android.os.Build.*` fields do not return null.